sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_client_async.py

```diff
@@ -897,13 +901,51 @@ async def _message_generator_async(self, timeout=None):
             if self._shutdown:
                 await self.close_async()
 
+    async def _on_disposition_received_async(self, message_delivery, reason, state):        
+        message_delivery.reason = reason
+        if reason == LinkDeliverySettleReason.DISPOSITION_RECEIVED:
+            if state and (SEND_DISPOSITION_ACCEPT in state or
+                SEND_DISPOSITION_RELEASE in state or
+                SEND_DISPOSITION_MODIFY in state or
+                SEND_DISPOSITION_RECEIVED in state or
+                (SEND_DISPOSITION_REJECT in state and state[SEND_DISPOSITION_REJECT][0] == None)):
```

Comment:
this particular one is interesting. Does the service return back a reject and no error message

`SEND_DISPOSITION_ACCEPT` would be the one to be the "we settled properly" state. Even the others mean that ?